lib/rules/no-component-lifecycle-hooks.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+
+const { isComponentLifecycleHook, isEmberComponent, isGlimmerComponent } = emberUtils;
+
+const ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS =
+  'Do not use classic ember components lifecycle hooks. Prefer using "@ember/render-modifiers" or custom functional modifiers.';
+
+const report = (context, node) => {
+  context.report(node, ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS);
+};
+
+module.exports = {
+  ERROR_MESSAGE_NO_COMPONENT_LIFECYCLE_HOOKS,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'disallow usage of "classic" ember component lifecycle hooks. Render modifiers or custom functional modifiers should be used instead.',
+      category: 'Ember Octane',
+      recommended: false,
+      octane: true,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-component-lifecycle-hooks.md',
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [],
+  },
+
+  create(context) {
+    let isInsideEmberComponent = false;
+    let currentEmberComponent = null;
+
+    const isAClassicLifecycleHook = keyName => {
+      return isComponentLifecycleHook(keyName) && isInsideEmberComponent;
+    };
+
+    return {
+      ClassDeclaration(node) {
+        isInsideEmberComponent =
+          isEmberComponent(context, node) || isGlimmerComponent(context, node);
```

Comment:
I believe you need to set `currentEmberComponent` in this case too, in case there is a class-inside-class situation.

Reply:
ah, yeah indeed. Added a test case and fix 👍